src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.ContentManagement.Display.ContentDisplay
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static ContentPartBuilder WithDisplayDriver<TContentPartDisplayDriver>(this ContentPartBuilder contentPartBuilder)
+            where TContentPartDisplayDriver : class, IContentPartDisplayDriver
+        {
+            // For backward compatability, probably not required.
+            //contentPartBuilder.Services.AddScoped<IContentPartDisplayDriver>(sp => sp.GetRequiredService<TContentPartDisplayDriver>());
+
+            var displayDriverType = typeof(TContentPartDisplayDriver);
+
+            if (displayDriverType.BaseType != null && !Array.Exists(displayDriverType.BaseType.GenericTypeArguments, x => x == contentPartBuilder.ContentPartType))
+            {
```

Comment:
Here just as i remember.

Maybe just `BaseType.GenericTypeArguments[0]`. Hmm, yes but if empty it would fail without your message.

So, what you do if `BaseType` is null? Do we need to check this? Is it possible to tied a part to a non generic driver?

Hmm, maybe (but it depends on the above)

    if (!BaseType.IsGenericType || BaseType.GenericTypeArguments[0] != ContentPartType)

With a null check on the `BaseType` would be `!(BaseType?.IsGenericType ?? false)`